src/main/resources/static/assets/js/sdh.js

```diff
@@ -275,8 +305,46 @@ class ModalEditRow {
 		            // Wrapped in $nextTick to ensure DOM is rendered before closing
 		            this.$refs.modal.hide();
 		        });
+		    },
+		    validateResponse() {
+		    	const blankAnswer = this.newResponse.questions.some(q => q.answer === "");
+		    	if (blankAnswer) {
+		    		alert("Please answer all questions");
+		    	} else {
+		    		this.submitResponse();
+		    	}
+		    },
+		    submitResponse() {
+		    	let obj = {
+		    		answerJson: JSON.stringify(this.newResponse.questions),
+		    		patient: `${this.contextPath}/api/patient/${this.patient}`,
+		    		sdhDomainQuestionnaire: this.newResponse.href
+		    	}
+				$.ajax({
+					url: `${this.contextPath}/api/questionnaire_response/`,
+					method: 'POST',
+					data: JSON.stringify(obj),
+					contentType: 'application/json',
+					success: data => {
+						$.ajax({
+							url: `${this.contextPath}/api/questionnaire_response/search/findByPatientId?id=${this.patient}`,
+							contentType: 'application/json',
+							success: data => {
+								this.responses = data._embedded.questionnaireResponses;
+								this.selectedResponseDate = this.currentResponse.createdAt;
+								this.tabIndex = 0;
+							}
+						});
+					}
+				});
+		    	
 		    }
 		},
+		watch: {
+			selectedResponseDate: function(newDate, oldDate) {
+				// This triggers the UI to update the selectedResponse
```

Comment:
It seems kind of weird to have a watcher that doesn't perform any actions of its own. Could this be replaced by a change handler on the field itself?

Reply:
This was one of the blockers I ran into. The change handler was always capturing the state previous to the change. I'm guessing this means there is some interaction in computed properties that is not quite right, but after many attempts I landed on using watch. I'd definitely like to revisit this one.

Reply:
I've definitely had some frustrations related to computed properties not updating as expected.

I'm not super familiar with the Bootstrap components you're using, but with a plain select list, the change handler would give you an event object that you could use to get the new selection (via event.target.value), but I'm surprised the v-model isn't doing the right thing. Have you tried using [`$nextTick`](https://vuejs.org/v2/api/#vm-nextTickhttps://vuejs.org/v2/api/#vm-nextTick) to wait for the next turn of the run loop? Sometimes that helps.

Reply:
I stumbled on $nextTick later in relation to something else and wondered if that would have worked in this case but didn't try it.